workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -1,32 +1,87 @@
 ﻿using Improbable.Gdk.Core;
-using Improbable.Gdk.Mobile;
 using UnityEngine;
-using UnityEngine.UI;
 
 namespace Playground
 {
+    [ExecuteInEditMode]
     public class ConnectionScreenController : MonoBehaviour
     {
         private const string HostIpPlayerPrefsKey = "SpatialOSHostIp";
+        private const int GuiPadding = 10;
+        private const float VerticalPositionRatio = 0.4f;
 
-        [SerializeField] private GameObject connectionPanel;
-        [SerializeField] private InputField ipAddressInput;
-        [SerializeField] private Button connectButton;
-        [SerializeField] private Text errorMessage;
         [SerializeField] private GameObject clientWorkerPrefab;
+        [SerializeField] private Font font;
+        [SerializeField] private float screenWidthFontRatio = 20;
 
-        private GameObject worker;
+        private bool showConnectionGui = true;
+        private bool isConnecting = false;
+        private string ipAddressText = null;
+        private string errorMessage = "";
 
-        private string IpAddress => ipAddressInput != null ? ipAddressInput.text : null;
+        private GameObject worker;
 
         public void Awake()
         {
-            ipAddressInput.text = PlayerPrefs.GetString(HostIpPlayerPrefsKey);
-            connectButton.onClick.AddListener(TryConnect);
+            ipAddressText = PlayerPrefs.GetString(HostIpPlayerPrefsKey);
+        }
+
+        public void OnGUI()
+        {
+            if (!showConnectionGui)
+            {
+                return;
+            }
+
+            using (new GUILayout.AreaScope(new Rect(
+                GuiPadding,
+                Screen.height * VerticalPositionRatio,
+                Screen.width - GuiPadding * 2,
+                Screen.height - GuiPadding * 2)))
+            {
+                var guiEnabled = GUI.enabled;
+
+                GUI.enabled = !isConnecting;
+                try
+                {
+                    using (new ResizedGui(font, screenWidthFontRatio,
```

Comment:
this is probably due to me not knowing UI stuff in Unity too well, but why do we instantiate the `ResizedGUI`and then later we only use static methods. do we need this?

Reply:
is it to initialize the font and fontsize?

Reply:
You're right about the font and fontsize initialization, but it's also for restoration.

The struct is doing a trick here similar to [EditorGUILayout.HorizontalScope](https://docs.unity3d.com/ScriptReference/EditorGUILayout.HorizontalScope.html) - within the context of this using scope, all of these `GUI.skin` objects will be modified to have different font size configurations. When the `using` block is over, the skin objects will be restored to original state.

The alternative would be to do something like this:

```c#
var oldLabelFontSize = GUI.skin.label.fontSize;
var oldTextFieldFontSize = GUI.skin.textField.fontSize;
var oldButtonFontSize = GUI.button.fontSize;
var oldTextAreaFontSize = GUI.textArea.fontSize;

GUI.skin.label.fontSize = wantedFontSize;
GUI.skin.textField.fontSize = wantedFontSize;
GUI.skin.button.fontSize = wantedFontSize;
GUI.skin.textArea.fontSize = wantedFontSize;

// do some GUI actions here with different font sizes

GUI.skin.label.fontSize = oldLabelFontSize;
GUI.skin.textField.fontSize = oldTextFieldFontSize;
GUI.skin.button.fontSize = oldButtonFontSize;
GUI.skin.textArea.fontSize = oldTextAreaFontSize;
```

The static methods are helpers to ensure the vertical sizing will be fixed - a side effect of wanting to do non-traditional font sizes, because Unity assumes every line will be 16 pixels high. Can probably find an alternative to the static methods - just saw "line height" configuration potential ( note to self: try https://docs.unity3d.com/ScriptReference/EditorGUIUtility-singleLineHeight.html  )

Reply:
Checked, it's read only :(